mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/state/StatefulSettingScreenTest.kt

```diff
@@ -0,0 +1,37 @@
+package ch.epfl.sdp.mobile.test.state
+
+import androidx.compose.ui.test.junit4.createComposeRule
+import androidx.compose.ui.test.onNodeWithTag
+import androidx.compose.ui.test.performClick
+import ch.epfl.sdp.mobile.application.authentication.AuthenticatedUser
+import ch.epfl.sdp.mobile.state.StatefulSettingsScreen
+import com.google.common.truth.Truth.assertThat
+import io.mockk.every
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import org.junit.Rule
+import org.junit.Test
+
+class StatefulSettingScreenTest {
+  @get:Rule val rule = createComposeRule()
+
+  @Test
+  fun given_SettingScreenLoaded_when_clickingOnEditProfileName_then_functionShouldBeCalled() =
+      runTest {
+    val user = mockk<AuthenticatedUser>()
+    every { user.name } returns "test"
+    every { user.email } returns "test"
+    every { user.emoji } returns "test"
+    every { user.uid } returns "test"
+    every { user.followed } returns false
+    var state = false
+
+    val mockFunction = { state = true }
+
+    rule.setContent { StatefulSettingsScreen(user, mockFunction) }
+
+    rule.onNodeWithTag("editProfileName").performClick()
```

Comment:
You should use content descriptions, not test tags.

Reply:
How can I use content description to pin down an Icon ?

Reply:
Something along the lines of :

```
val strings = rule.setContentWithLocalizedStrings { Icon(..., contentDescription = LocalLocalizedStrings.current.something) )}
rule.onNodeWithContentDescription(strings.something).performClick()
```

Reply:
Could you please copy the stack trace from the test here ?

There could be many causes for this problem. For instance, it's possible that multiple nodes match the content description if it's not specific enough for instance.